File: Cpp/2834_find_the_minimum_possible_sum_of_a_beautiful_array.cpp
// https://leetcode.cn/problems/find-the-minimum-possible-sum-of-a-beautiful-array/
        
class Solution {
public:
    int minimumPossibleSum(int n, int target) {
        int half = target / 2, mod = 1e9+7;
        int res = 0;
        if (n <= half) {
            res = (((long long)1+n) * n / 2) % mod;
        }
        else {
            res = (((long long)1+half) * half / 2 +
                   ((long long)target+target+n-half-1) * (n-half) / 2 ) % mod;
        }
        return res;
    }
};

File: Python3/2834_find_the_minimum_possible_sum_of_a_beautiful_array.py
# https://leetcode.cn/problems/find-the-minimum-possible-sum-of-a-beautiful-array/

class Solution:
    def minimumPossibleSum(self, n: int, target: int) -> int:
        mod = int(1e9) + 7
        half = target // 2
        if n <= half:
            res = ((1 + n) * n // 2) % mod
        else:
            res = ((1 + half) * half // 2) % mod
            remain = n - half
            res = (res + (target + target + remain - 1) * remain // 2) % mod
        return res